Cloudscale server fixes

gaudenz opened the pull request:
##### SUMMARY
Fixes to the cloudscale_server module to improve various corner cases.

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
cloudscale_server

##### ANSIBLE VERSION
devel

gaudenz committed:
Improve error message in cloudscale_server module

Fix punctuation and add the full contents of "info" to the output in
case of failed API calls. This is useful in case of connection timeouts
and other error conditions where there is no response body available.

gaudenz committed:
Increase timeouts in cloudscale_server module

Increase the timeouts to not fail in case the API calls take a bit
longer than usual. The default timeout of fetch_url is 10s which is
quite short. Increase it to 30s. The timeout for waiting for a server
change is increased as well as it calls the API in a loop. Therefore
this value should be larger than the API timeout.

gaudenz committed:
Send API parameters as JSON in cloudscale_server module

Use JSON to send the POST data to the API instead of an urlencoded
string. Urlencoding is not really a good match for some Python
datatypes.

This fixes an issue when submitting a list of SSH keys which did not get
translated properly.

resmo commented on lib/ansible/modules/cloud/cloudscale/cloudscale_server.py:
```diff
@@ -203,12 +203,12 @@
 API_URL = 'https://api.cloudscale.ch/v1/'
-TIMEOUT_WAIT = 30
+TIMEOUT_WAIT = 60
+TIMEOUT_API = 30
```
Because it probably depends on the user's network, would it help to let the user overwrite the timeout by a param e.g. `api_timeout`, so these values are only used as default?

resmo reviewed (approved):
shipit

gaudenz committed:
Fix typo in cloudscale_server documentation

gaudenz committed:
cloudscale_sever: Replace timeout const by api_timeout param

Replace the static TIMEOUT_API constant by a user configurable
api_timeout parameter. Also eliminate the TIMEOUT_WAIT constant by
2*api_timeout. This means that the timeout to wait for server changes is
always double the timeout for API calls.

gaudenz committed:
Use Debian 9 image for cloudscale_server tests

gaudenz commented:
@resmo I pushed some more commits which address your concern.

resmo commented:
Thanks!
